jetty-util-ajax/src/main/java/org/eclipse/jetty/util/ajax/AsyncJSON.java

```diff
@@ -0,0 +1,1205 @@
+            switch (peek)
+            {
+                case '"':
+                {
+                    if (stack.peek().state == State.OBJECT_FIELD)
+                    {
+                        stack.push(State.OBJECT_FIELD_NAME, UNSET);
+                        if (parseString(buffer))
+                        {
+                            // We are not done yet, parse until the ':'.
+                            break;
+                        }
+                        return false;
+                    }
+                    else
+                    {
+                        throw newInvalidJSON(buffer, "invalid object field");
+                    }
+                }
+                case ':':
+                {
+                    buffer.get();
+                    // We are done with the field name.
+                    String fieldName = (String)stack.peek().value;
+                    stack.pop();
+                    // Change state to parse the field value.
+                    stack.push(fieldName, State.OBJECT_FIELD_VALUE, UNSET);
+                    return true;
+                }
+                default:
+                {
+                    if (Character.isWhitespace(peek))
+                    {
+                        buffer.get();
+                        break;
+                    }
+                    else
+                    {
+                        throw newInvalidJSON(buffer, "invalid object field");
+                    }
+                }
+            }
+        }
+        return false;
+    }
+
+    private boolean parseObjectFieldValue(ByteBuffer buffer)
+    {
+        if (stack.peek().value == UNSET)
+        {
+            if (!parseAny(buffer))
+                return false;
+        }
+
+        // We are done with the field value.
+        Frame frame = stack.peek();
+        Object value = frame.value;
+        String name = frame.name;
+        stack.pop();
+        // We are done with the field.
+        stack.pop();
+        @SuppressWarnings("unchecked")
+        Map<String, Object> map = (Map<String, Object>)stack.peek().value;
+        map.put(name, value);
+
+        return true;
+    }
+
+    private Object convertObject(Map<String, Object> object)
+    {
+        Object result = convertObject("x-class", object);
+        if (result == null)
+        {
+            result = convertObject("class", object);
+            if (result == null)
+                return object;
+        }
+        return result;
+    }
+
+    private Object convertObject(String fieldName, Map<String, Object> object)
+    {
+        String className = (String)object.get(fieldName);
+        if (className == null)
+            return null;
+
+        JSON.Convertible convertible = toConvertible(className);
+        if (convertible != null)
+        {
+            convertible.fromJSON(object);
+            return convertible;
+        }
+
+        JSON.Convertor convertor = factory.convertors.get(className);
```

Comment:
This will NPE if no convertor was registered

Reply:
I added a `getConvertor(String)` in the `Factory` and use it here.